### PR TITLE
feat(widgets/chart): add option to set the position of legend

### PR DESCRIPTION
**Pull request by lyuha**
Migerated PR from tui-rs is #25

Add option to position the legend at corner. The option uses tui::layout::Corner emum that already made.
If the layout have axis title, the layout draw the legend position above x axis title or under y axis title.

Examples all positions of legend
ratatui-chart-legend

I make 3 tests for each corners of legend based on original test for legend.

### CODE REVIEW

**Review by joshka — CHANGES_REQUESTED** [review]
I'd suggest also adding / updating the existing [chart example](https://github.com/ratatui-org/ratatui/tree/main/examples#chart-chartrs) to show the legend positioning:
image

File: src/widgets/chart.rs
Comment (lyuha) [reply src/widgets/chart.rs]: Update this code to use smaller size of both.

**Comment by lyuha** [comment]
I'll fix CI and add more test codes. And I make commits cumulativly.
Before merge, I will rebase and squash commits for clean git history.

**Comment by lyuha** [comment]
CI failed because of `codecov.io`. Please check CI pass.

```
[2023-08-21T15:53:03.740Z] ['error'] There was an error running the uploader: Error uploading to [https://codecov.io:](https://codecov.io/) Error: There was an error fetching the storage URL during POST: 404 - {'detail': ErrorDetail(string='Unable to locate build via Github Actions API. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}
```

**Comment by lyuha** [comment]
Update chart examples all code requested changes.

chart example

At this time, rebase to main branch, CI failed. But CI failed due to an other reason.

Here is error.
```console
Can't use 'tar -xzf' extract archive file: /home/runner/work/_actions/_temp_fcc1cf2d-bf9a-43b0-8047-ca17e1e0339a/f92c391d-66ef-45d5-9351-73185dd01cc5.tar.gz
```

**Comment by lyuha** [comment]
Rebase to main branch a7bf4b3.
Passed CI!

File: examples/chart.rs
Comment (Valentin271) [inline examples/chart.rs:258]: This doesn't show up on the example because charts are too smalls (at least on my screen), plus it shows the default position.

What do you think about doing:
- First chart stays the same
- Loosen `hidden_legend_constraints` so we can see the default legend on the second chart
-  Third chart, use non default position and loosen constraints to see it

Something like:
image

**Review by Valentin271 — CHANGES_REQUESTED** [review]
Mainly this needs documentation.
Also could you rebase on main, this PR has been hanging for quite a while now ...

File: src/widgets/chart.rs
Comment (lyuha) [reply src/widgets/chart.rs:322]: [b6e6c0d](https://github.com/ratatui-org/ratatui/pull/378/commits/b6e6c0d30f78a129340695026885596a914b1803) add description about `None` position into `legend_position` function. Because `legend_position` is private property.

File: src/widgets/chart.rs
Comment (lyuha) [reply src/widgets/chart.rs]: Modified it! [210a291 Diff](https://github.com/ratatui-org/ratatui/pull/378/commits/210a291eeecabc04cb9048a535b6db7db45f1722#diff-657af648164c658c906487fd1975fc7d684fe53ae5dc035a88ea864d88333919L333-R334)

File: src/widgets/chart.rs
Comment (Valentin271) [reply src/widgets/chart.rs:322]: Works for me :+1:

**Review by Valentin271 — APPROVED** [review]
Looks good, thank you for your contribution!

**Comment by Valentin271** [comment]
Just missing an approval from @joshka

**Review by orhun — APPROVED** [review]

**Review by joshka — APPROVED** [review]

**Pull request merged**
